scipy/optimize/_lsq/lsq_linear.py

```diff
@@ -32,7 +32,8 @@ def prepare_bounds(bounds, n):
 
 
 def lsq_linear(A, b, bounds=(-np.inf, np.inf), method='trf', tol=1e-10,
-               lsq_solver=None, lsmr_tol=None, max_iter=None, verbose=0):
+               lsq_solver=None, lsmr_tol=None, lsmr_max_iter=None,
+               max_iter=None, verbose=0):
```

Comment:
I think the new `lsmr_max_iter` argument will need to be added at the end of the parameter list for backwards compatibility of the positional-only incantation.